client.go

```diff
@@ -224,27 +275,33 @@ func (c *Client) checkResponse(resp *http.Response, err error) (*http.Response,
 	if err != nil {
 		return resp, fmt.Errorf("Error calling the API endpoint: %v", err)
 	}
-	if 199 >= resp.StatusCode || 300 <= resp.StatusCode {
-		var eo *errorObject
-		var getErr error
-		if eo, getErr = c.getErrorFromResponse(resp); getErr != nil {
-			return resp, fmt.Errorf("Response did not contain formatted error: %s. HTTP response code: %v. Raw response: %+v", getErr, resp.StatusCode, resp)
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		// check whether the error response is declared as JSON
+		if !strings.HasPrefix(resp.Header.Get("Content-Type"), "application/json") {
+			return resp, fmt.Errorf("HTTP response with status code %d does not contain Content-Type: application/json", resp.StatusCode)
+		}
```

Comment:
Wondering if there are versions of the API that respond with JSON but don't bother setting this header? Or, are there versions of the API that respond with a 2xx status, but have a JSON-encoded error message?

If yes, is it worth caring about?

Reply:
In a past life I worked at PagerDuty, and at least when I was there and helped design a version of the API it was important that the `Content-Type` was set and correct. I think we should feel comfortable that will be set when they return JSON from the API (whether success or an error)

I'm expecting their Nginx load balancers may not provide a JSON response if they fail, and so this would not try to parse any sort of junk that may spit back but provide the status code. In that case I'd expect it to be `5XX`, which most consumers would interpret to mean that PagerDuty experienced some sort of failure that resulted in a malformed response. So I think this is okay.

Their documentation indicates that 2XX codes will not use an error JSON document:

> If an API request is successful, PagerDuty will respond to it with HTTP status code 2XX and the resources that match the query.
> 
> If the API request is unsuccessful, PagerDuty will respond with a corresponding HTTP error status code and the response body will contain PagerDuty error details.
- https://developer.pagerduty.com/docs/rest-api-v2/errors/